src/main/java/org/jabref/logic/formatter/bibtexfields/NormalizeMonthFormatter.java

```diff
@@ -23,6 +26,31 @@ public String getKey() {
     public String format(String value) {
         Objects.requireNonNull(value);
         Optional<Month> month = Month.parse(value);
+
+        if (!month.isPresent()) {
+            // Normalize month name to unicode, which is returned by DateFormatSymbols.getMonths() - see below
+            value = new LatexToUnicodeAdapter().format(value);
+
+            if (value.equalsIgnoreCase("Maerz")) {
+                // this value is not treated by LatexToUnicodeAdapter, so a special handling is required.
+                month = Month.getMonthByNumber(3);
+            } else {
```

Comment:
Why don't you use the java 8 DateTimeFormatter for parsing the dates? Much more easier and flexible https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html#predefined